doc/ADR/release_process.md

```diff
@@ -63,6 +80,24 @@ detectable change including bugfixes can do that. Instead, major version
 increments should signal that we expect many users will have to review their
 workflows for potential breakages, not just any.
 
+### Rust API versioning
+
+Following discussions during the development of `libcprover_rust`, it was decided
+that it would be easiest if the Rust API version was following the same scheme
+that CBMC does, so that it's clear which version of the Rust API corresponds to
+which version of CBMC.
+
+This means that the Rust API has two "versions" it can report.
+
+1. The library version, as reported in `Cargo.toml`, and
+2. The (C++) API version, which is coming from CBMC and aligns with the version
+   number in `config.inc` (reported via the `get_api_version()` API call).
+
+Most of the time, these two numbers will be the same - but there can be scenarios
+where they can diverge: if the user for example uses a later version of the
+Rust API (say, `5.80.0`) to link against a static library representing an older
+version of CBMC (say `5.79.0`).
```

Comment:
Outside the scope of this PR, but should this be allowed or prevented? Are there times when this is safe and times when it could yield problematic outcomes?

Reply:
The Rust API needs to link against `libcprover.x.y.z.a` to use the functions exposed by the C++ API.

If the C++ API functions haven't been subjected to an interface change for the existing functions, then this should be safe.

Of course, there cannot be too wide a divergence between versions of the Rust and C++ API it depends on.

In my personal experiments, I have linked the Rust API version 5.80.0 against `libcprover.5.78.0.a` and `libcprover.5.79.0.a` and everything worked fine. But of course, there's no guarantee that even a weekly release change is going to work.

I can try linking it against an even earlier version to see what happens, but based on behaviour I have observed before, if this is not possible, the most likely outcome is a linking error.